sklearn/manifold/tests/test_isomap.py

```diff
@@ -185,4 +185,19 @@ def test_sparse_input():
             clf = manifold.Isomap(n_components=2,
                                   eigen_solver=eigen_solver,
                                   path_method=path_method)
-            clf.fit(X)
+            if eigen_solver != 'randomized':
+                # nominal scenario: it should work
+                clf.fit(X)
+            else:
+                # the randomized method is still inconsistent with the others
+                # on this case: if the gram matrix sent by isomap contains
+                # large negative eigenvalues they can be selected instead of
+                # the 2 largest positive. In that case there is a protection in
+                # KernelPCA: a ValueError is raised instead of using the wrong
+                # components
+                try:
```

Comment:
Can we set the random state of `sparse_rand` so we can above this code path?

Reply:
I found that with some random states, some methods work, but not all. For clarity I will create a dedicated test

Reply:
However apparently I cannot get consistent results across all CI nodes, so even with a dedicated test, I cannot rely on the random state to get deterministic success/fail. I'll revert this new test back to a try/except